src/lib/features/metrics/instance/instance-service.ts

```diff
@@ -224,4 +225,20 @@ export default class ClientInstanceService {
     async removeInstancesOlderThanTwoDays(): Promise<void> {
         return this.clientInstanceStore.removeInstancesOlderThanTwoDays();
     }
+
+    async usesSdkOlderThan(
+        sdkName: string,
+        sdkVersion: string,
+    ): Promise<boolean> {
+        const semver = new SemVer(sdkVersion);
+        const instancesOfSdk =
+            await this.clientInstanceStore.getBySdkName(sdkName);
+        return instancesOfSdk.some((instance) => {
+            if (instance.sdkVersion) {
+                const [_sdkName, sdkVersion] = instance.sdkVersion.split(':');
+                const instanceUsedSemver = new SemVer(sdkVersion);
+                return instanceUsedSemver < semver;
+            }
+        });
```

Comment:
I was just checking that SemVer can throw some exceptions: https://github.com/npm/node-semver/blob/main/classes/semver.js and we have a helper class that does the try catch: https://github.com/Unleash/unleash/blob/chore/update-features-created-by-user-id/src/lib/util/semver.ts maybe we should just try catch here

Reply:
Thanks. Good point. I'll have a look